cmd/restic/cmd_rewrite.go

```diff
@@ -217,6 +217,31 @@ func filterAndReplaceSnapshot(ctx context.Context, repo restic.Repository, sn *r
 	sn.Original = sn.ID()
 	sn.Tree = &filteredTree
 
+	sn.Summary.TotalFilesProcessed = 0
```

Comment:
For old snapshots, summary is a nil pointer, such that this would cause a crash.